test/e2e/create_test.go

```diff
@@ -314,11 +365,28 @@ func (tc *testContext) testBYOHConfiguration(t *testing.T) {
 		err = tc.provisionBYOHConfigMapWithMachineSet()
 		require.NoError(t, err, "error provisioning BYOH ConfigMap with MachineSets")
 	}
+	configMapCreatedTime := time.Now()
 	// Wait for Windows worker node to become available
 	t.Run("VM is configured by ConfigMap controller", func(t *testing.T) {
 		err := tc.waitForConfiguredWindowsNodes(gc.numberOfBYOHNodes, false, true)
 		assert.NoError(t, err, "Windows node creation failed")
 	})
+
+	// Save timings as an artifact for performance measuring purposes
+	configTimes := configurationTimes{
+		NodeType:       "byoh",
+		NodeCount:      int(gc.numberOfBYOHNodes),
+		NodeConfigured: time.Now().Sub(configMapCreatedTime).Round(time.Second).String(),
+	}
+	times, err := json.Marshal(configTimes)
+	require.NoError(t, err)
+	times = append(times, '\n')
+	fd, err := os.OpenFile(filepath.Join(tc.artifactDir, "config-time.json"), os.O_WRONLY|os.O_CREATE|os.O_APPEND, os.ModePerm)
+	require.NoError(t, err)
+	defer fd.Close()
+	_, err = fd.Write(times)
+	require.NoError(t, err)
```

Comment:
Consider generalizing this block to avoid duplication, so that the name of the resulting file (config-time.json) is defined only once.